src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -73,6 +72,7 @@ class PgAutoscaler(MgrModule):
     def __init__(self, *args, **kwargs):
         super(PgAutoscaler, self).__init__(*args, **kwargs)
         self._shutdown = threading.Event()
+        self._event = {}
```

Comment:
Events within the progress module are stored in the persistent key value store of the progress module. `self._event = {}` is not stored permanently. 

Is there a possibility this can lead to progress events without references to them form the autoscaler module due to MGR restart?

Reply:
@sebastian-philipp 
Hi,
so I the purpose of  `this self._event = {}` is just for local references to track if the is an autoscaler event at a pool already, so we don't need to create a new one, but rather update it. But yes basically when we craete an event locally, we also create one in the progress module as well. There is an alternative strategy where we could just reference everything in the progress event so like we could check the events in the progress and see which pool it is referencing to, therefore we don't another `this._event` in the pg_autoscaler.

Reply:
> @sebastian-philipp
> Hi,
> so the purpose of `this self._event = {}` is just for local references to track if the is an autoscaler event at a pool already, so we don't need to create a new one, but rather update it. But yes basically when we craete an event locally, we also create one in the progress module as well. 

Just run a `ceph mgr fail $(ceph mgr dump | jq -r '.active_name')` while a pg_autoscaler event is there and unfinished. The progress module should then reload all events from the store.

Reply:
I guess we'd need to stash and repopulate this module's events too - could that metadata (initial_pg_num/pg_num_target) be added to the remote event and handled by the existing progress module persistence?